### PR TITLE
Chefgpt weekend homework

### PR DESCRIPTION
**Pull request by ionescu77**
## Task

Modify and expand the Chef GPT script by incorporating a unique personality for your AI chef
Tweak the system prompt to include a unique personality for your AI chef
Example personalities:
- [x] A young, enthusiastic Indian chef specializing in Biryani
- [x] A seasoned Italian chef with a passion for pasta-making

@alexmazaltov I have implemented the indian chef.
Can you implement the italian chef and push to this PR?

### CODE REVIEW

**Comment by alexmazaltov** [comment]
Yes i can.

**Comment by alexmazaltov** [comment]
@ionescu77 Please review my contribution and Merge so we can proceed with the following:

## Conduct a comprehensive experiment:
- The first person should use one script with one personality to suggest a dish based on given ingredients
- After running the script, send the response for one of your group members (via Discord or any other means)
- The second person should request a recipe for that dish using a second script with a different personality
- After running the script, send the response for another of your group members
- The third person should critique the provided recipe using a third script with a different personality

## Compile a simple report documenting:
- The experiment process
- The system prompts used in each script
- Comparative analysis of the different user prompts and their responses

**Comment by ionescu77** [comment]
Cool, let me pull the changes and test on my laptop and will get back to you.

**Comment by ionescu77** [comment]
## Conclusions

I run the script choosing italian personality.

- the script correctly refused recipe for cocktail `gin tonic`
- the script correctly refused another local dish `sarmale`
- the script provided correct recipe for personality

## Output
```bash
python chef-gpt.py
Type the name of the dish you want a recipe for:
blabla
Choose your chef (1 for experienced chef, 2 for Italian pasta expert):
2
```
Mi dispiace, but I don't recognize "blabla." If you have a specific Italian dish or recipe in mind, please let me know, and I'll be more than happy to help you with a detailed recipe. Your passion for good food deserves the best!
If you have any other questions, please type them here:
```
gin tonic
Choose your chef (1 for experienced chef, 2 for Italian pasta expert):
2
```
Mi dispiace, but I don't recognize "gin tonic" as an Italian dish. If you're looking for a specific Italian recipe or pasta dish, please let me know! I'm here to help you whip up something delicious, but it must be rooted in the culinary traditions of Italy! 🍝
If you have any other questions, please type them here:
```
pizza
Choose your chef (1 for experienced chef, 2 for Italian pasta expert):
2
```
Ah, la pizza! An iconic Italian dish that captures the hearts (and stomachs) of many! Let me guide you through a detailed recipe for crafting your very own classic Margherita pizza. It's simple yet divine, with the beauty of its ingredients shining through.

### Classic Margherita Pizza Recipe

#### Ingredients

**For the Dough:**
- 4 cups (500g) all-purpose flour (plus extra for dusting)
- 1 ½ teaspoons salt
- 1 teaspoon sugar
- 1 packet (2 ¼ teaspoons) active dry yeast
- 1 ½ cups (360 ml) warm water (about 110°F or 43°C)
- 1 tablespoon extra virgin olive oil

**For the Toppings:**
- 1 can (14 oz/400g) San Marzano tomatoes (or any good quality canned tomatoes)
- 8 oz (225g) fresh mozzarella cheese, sliced
- Fresh basil leaves
- Salt, to taste
- 2 tablespoons extra virgin olive oil (for drizzling)

#### Equipment
- Pizza stone or baking sheet
- Mixing bowl
- Rolling pin
- Parchment paper

### Instructions

#### Step 1: Prepare the Dough
1. **Activate the Yeast:** In a small bowl, combine warm water and sugar. Sprinkle the yeast on top and let it sit for about 5 minutes until it becomes frothy.

2. **Mix the Ingredients:** In a large mixing bowl, combine flour and salt. Create a well in the center and pour in the yeast mixture along with the olive oil.

3. **Knead the Dough:** Mix the ingredients until they form a shaggy dough, then transfer it to a floured surface. Knead the dough for about 8-10 minutes until it's smooth and elastic. (Oh, the joy of kneading! It’s a workout and a meditation!)

4. **First Rise:** Place the dough in a lightly oiled bowl, cover it with a damp cloth, and let it rise in a warm place for about 1-2 hours, or until it doubles in size. Patience, amico! Good things come to those who wait!

#### Step 2: Prepare the Sauce
1. **Make the Sauce:** In a bowl, crush the canned tomatoes by hand or with a fork. Add a pinch of salt to taste. (Simplicity is key; we want that fresh tomato flavor to shine!)

#### Step 3: Shape and Preheat
1. **Preheat the Oven:** Preheat your oven to its highest setting, about 475°F (245°C). If using a pizza stone, place it in the oven while it’s preheating.

2. **Shape the Pizza:** After the dough has risen, punch it down and divide it into two equal pieces (or more if you prefer individual pizzas!). Roll each piece into a ball, cover, and let them rest for about 30 minutes.

3. **Form the Base:** On a floured surface, stretch or roll out one dough ball into a circle about 12 inches (30 cm) in diameter. Keep the edges slightly thicker for that lovely crust!

#### Step 4: Assemble and Bake
1. **Add the Toppings:** Transfer the rolled-out dough onto a piece of parchment paper. Spread a thin layer of your tomato sauce on the base. Add slices of fresh mozzarella evenly. Tear some fresh basil leaves and scatter them on top.

2. **Bake the Pizza:** Carefully transfer the pizza (with the parchment) onto the hot pizza stone or baking sheet. Bake for about 8-10 minutes, or until the crust is golden brown and the cheese is bubbly. Perfetto!

3. **Finishing Touches:** Once out of the oven, drizzle with a little extra virgin olive oil and add more fresh basil if desired.

#### Step 5: Serve
1. **Slice and Enjoy:** Let the pizza cool for a minute before slicing. Serve hot, and enjoy this beautiful creation you’ve made!

Buon appetito! 🍕✨ Remember, each pizza is a canvas for your creativity—experiment with toppings while honoring the essence of traditional Italian pizza-making! If you have any questions, don't hesitate to ask!
If you have any other questions, please type them here:

File: chef-gpt-indian.py
Comment (ionescu77) [inline chef-gpt-indian.py:1]: No changes here by team

File: chef-gpt.py
Comment (ionescu77) [inline chef-gpt.py:1]: Interesting implementation using python logic.

File: chef-gpt.py [hunk 1]
```diff
@@ -21,20 +25,25 @@
...
+if chef_personality == "1":
+    personality = "experienced chef"
+else:
+    personality = "Italian pasta expert"
+
 messages.append(
     {
         "role": "user",
-        "content": f"Suggest me a detailed recipe and the preparation steps for making {dish}"
+        "content": f"As the {personality}, suggest me a detailed recipe and the preparation steps for making {dish}"
```
Comment (ionescu77) [inline chef-gpt.py:46]: Here we will feed in the request one of the two personalities choosen by the user.

**Pull request merged**